src/hooks/sprint-custom-hooks.ts

```diff
@@ -0,0 +1,93 @@
+import type { SelectChangeEvent } from "@mui/material";
+import { useState } from "react";
+import type { ResourceAllocations, ResourceAllocationsProject } from "src/generated/homeLambdasClient";
+
+const useSprintViewHandlers = () => {
+  const [filterType, setFilterType] = useState("project");
+  const [searchQuery, setSearchQuery] = useState("");
+  const [selectedProject, setSelectedProject] = useState<ResourceAllocationsProject | null>(null);
+
+  /**
+   * Handles the change event for the filter selection.
+   * 
+   * @param event - The event triggered by the filter selection change.
+   */
+  const handleFilterChange = (event: SelectChangeEvent<string>) => {
+    setFilterType(event.target.value);
+    setSearchQuery("");
+    setSelectedProject(null);
+  };
+
+  /**
+   * Handles the click event on a row in the resource allocations table.
+   * 
+   * @param row - type of ResourceAllocations.
+   */
+  const handleRowClick = (row: ResourceAllocations) => {
+    if (filterType === "project") {
+      setSearchQuery(row.project?.name || "");
+      setSelectedProject(row.project || null);
+    } else if (filterType === "user") {
```

Comment:
We could use the enum I mentioned in a previous comment here. It's good to use enums in cases like this, easier typing and reduces the change for human error